From 690bee3feb9a4bb290f62255483a646b9aa7b23b Mon Sep 17 00:00:00 2001
From: Matt Fleming <matt.fleming@intel.com>
Date: Mon, 6 Oct 2014 14:09:10 +0100
Subject: [PATCH] efi_runtime: Don't use get_user() on non-pointer

We don't need to try to construct a pointer from a structure on the
stack in order to use get_user(). We can just access it directly.

This fixes an efi_runtime_set_waketime() failure, introduced with commit
"fwts: Copied the structure from userland locally in kernel space".

Signed-off-by: Matt Fleming <matt.fleming@intel.com>
---
 efi_runtime/efi_runtime.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/efi_runtime/efi_runtime.c b/efi_runtime/efi_runtime.c
index 249f2619ca23..7332d6b090eb 100644
--- a/efi_runtime/efi_runtime.c
+++ b/efi_runtime/efi_runtime.c
@@ -397,8 +397,8 @@ static long efi_runtime_set_waketime(unsigned long arg)
 	if (copy_from_user(&psetwakeuptime_local, psetwakeuptime, sizeof(psetwakeuptime_local)))
 		return -EFAULT;
 	
-	if (get_user(enabled, &(psetwakeuptime_local.Enabled)) || 
-				copy_from_user(&efi_time, psetwakeuptime_local.Time, sizeof(EFI_TIME)))
+	enabled = psetwakeuptime_local.Enabled;
+	if (copy_from_user(&efi_time, psetwakeuptime_local.Time, sizeof(EFI_TIME)))
 		return -EFAULT;
 	
 	convert_to_efi_time(&eft, &efi_time);
-- 
1.9.3

